superduperdb/backends/mongodb/query.py

```diff
@@ -394,9 +394,23 @@ def add_fold(self, fold: str):
     @applies_to('insert_many', 'insert_one')
     def documents(self):
         """Return the documents to insert."""
+
+        def _wrap_document(document):
+            if not isinstance(document, Document):
+                if isinstance(document, dict):
+                    document = Document(document)
+                else:
+                    document = Document({'_base': document})
+            return document
+
         if self.parts[0][0] == 'insert_many':
-            return self.parts[0][1][0]
-        return [self.parts[0][1][0]]
+            documents = self.parts[0][1][0]
+            wrapped_documents = []
+            for document in documents:
+                document = _wrap_document(document)
```

Comment:
After constructing the new document, should we assign it back to query.parts[0][1][0]? Otherwise, modifications made to the document elsewhere might not take effect here. Should we allow it to update accordingly?

WDYT?

Reply:
Done, discussed with @blythed !